### DISCUSSION TITLE
Auto completion for note-titles with spaces

### DISCUSSION THREAD

**Discussion by vicrdguez** [zk-org/zk-nvim#135] [Help] [answered]
Hi!

First of all thanks for the plug-in and the cli tool. It's great!

One pain point I have is when using in-line completions, writing a space breaks the completion making it imposible to link to my notes. 

Additionally I have notes with rather long and descriptive titles so the issue becomes more noticeable. 

Of course I can always insert links by using the provided function that launches telescope in my case but:

- After the link is added, you are not in Insert mode, and the cursor is placed out of the link so you need to go back if you want to change the link name (for aliases e.g). Minor but is there

- Telescope default display mode takes the whole screen space, and that breaks a lot the writing flow. I've went ahead and changes light my the plug-in code locally to use Ivy, but definitely I prefer not to make changes to plugins like that. 

How do you guys do it? How do you configure nvim-cmp so the space don't break the completion?

**Comment by mickael-menu** [comment, marked as answer]
> One pain point I have is when using in-line completions, writing a space breaks the completion making it imposible to link to my notes.

That's a limitation of the LSP client (e.g. Neovim). Usually they only auto-complete tokens so a space interrupts the completion. However you don't need to add spaces, you can just write everything attached and even omit part of the words:

<img width="458" alt="Screenshot 2023-06-24 at 18 03 23" src="https://github.com/mickael-menu/zk-nvim/assets/58686775/8e965bc8-64ec-487f-b9a6-f343fe14feec">

> Telescope default display mode takes the whole screen space, and that breaks a lot the writing flow. I've went ahead and changes light my the plug-in code locally to use Ivy, but definitely I prefer not to make changes to plugins like that.

I *think* you should be able to provide your own Telescope options by passing a `telescope` table of options in the `zk` APIs options.

https://github.com/mickael-menu/zk-nvim/blob/5ddb53688035d115f941f0c8255f6e6618e608ac/lua/zk/pickers/telescope.lua#L62

**Reply by vicrdguez** [reply]
Hi! sorry I missed this one.

Thank for the suggestion, yes I've been researching a bit about it and looks that is a limitation of the client itself. I fell back to use the insert link command for mappings for both normal and insert modes. I was using the same when I was using emacs so it feels familiar. 

It comes with its own quirks though, since the the cursor remains before the inserted link instead of being placed after, but not a big deal, I'll look into the code for a potential solution when I have the time.
